Algolia.Search/Index.cs

```diff
@@ -172,6 +172,37 @@ public Task<JObject> GetObjectsAsync(IEnumerable<String> objectIDs, Cancellation
         }
 
         /// <summary>
+        /// Get several objects from this index.
+        /// </summary>
+        /// <param name="objectIDs">An array of unique identifiers of the objects to retrieve.</param>
+        /// <returns></returns> 
+        public Task<JObject> GetObjectsWithAttributesAsync(IEnumerable<String> objectIDs, IEnumerable<string> attributesToRetrieve, CancellationToken token = default(CancellationToken))
+        {
+            JArray requests = new JArray();
+            var attributes = "";
+            foreach (string attr in attributesToRetrieve)
+            {
+                if (attributes.Length > 0) {
+                    attributes += ",";
+                    attributes += Uri.EscapeDataString(attr);

```

Comment:
Why do you encode the attribute names? We pass it in the body so we shouldn't need to encode it.